[AUTOCUT] Gradle Check Flaky Test Report for SearchIT

## Flaky Test Report for `SearchIT`

Noticed the `SearchIT` has some flaky, failing tests that failed during **post-merge actions**.

### Details

| Git Reference | Merged Pull Request | Build Details | Test Name |
|---------------|----------------------|---------------|-----------|
| 22a6194ee26d54a10a24b4b7c011a738a10c43c3 | [18124](https://github.com/opensearch-project/OpenSearch/pull/18124) | [58554](https://build.ci.opensearch.org/job/gradle-check/58554/testReport/) | `org.opensearch.client.SearchIT.testSearchWithMatrixStats` |
| 2c1b2aa6ee42d0381628f08cb4bf907c959a7934 | [18247](https://github.com/opensearch-project/OpenSearch/pull/18247) | [57899](https://build.ci.opensearch.org/job/gradle-check/57899/testReport/) | `org.opensearch.client.SearchIT.testSearchWithMatrixStats` |
| 83a4beb15aaa2391a9700e2dad827fa1cc340d87 | [18073](https://github.com/opensearch-project/OpenSearch/pull/18073) | [57316](https://build.ci.opensearch.org/job/gradle-check/57316/testReport/) | `org.opensearch.client.SearchIT.testSearchWithMatrixStats` |

The other pull requests, besides those involved in post-merge actions, that contain failing tests with the `SearchIT` class are:

- [14443](https://github.com/opensearch-project/OpenSearch/pull/14443)
- [15258](https://github.com/opensearch-project/OpenSearch/pull/15258)
- [16294](https://github.com/opensearch-project/OpenSearch/pull/16294)
- [17507](https://github.com/opensearch-project/OpenSearch/pull/17507)
- [16421](https://github.com/opensearch-project/OpenSearch/pull/16421)
- [18332](https://github.com/opensearch-project/OpenSearch/pull/18332)
- [16607](https://github.com/opensearch-project/OpenSearch/pull/16607)
- [17275](https://github.com/opensearch-project/OpenSearch/pull/17275)
- [16366](https://github.com/opensearch-project/OpenSearch/pull/16366)
- [16378](https://github.com/opensearch-project/OpenSearch/pull/16378)
- [17457](https://github.com/opensearch-project/OpenSearch/pull/17457)
- [17909](https://github.com/opensearch-project/OpenSearch/pull/17909)
- [17611](https://github.com/opensearch-project/OpenSearch/pull/17611)
- [17446](https://github.com/opensearch-project/OpenSearch/pull/17446)
- [17817](https://github.com/opensearch-project/OpenSearch/pull/17817)
- [18230](https://github.com/opensearch-project/OpenSearch/pull/18230)
- [18194](https://github.com/opensearch-project/OpenSearch/pull/18194)
- [16891](https://github.com/opensearch-project/OpenSearch/pull/16891)
- [17961](https://github.com/opensearch-project/OpenSearch/pull/17961)
- [18333](https://github.com/opensearch-project/OpenSearch/pull/18333)
- [17538](https://github.com/opensearch-project/OpenSearch/pull/17538)
- [17890](https://github.com/opensearch-project/OpenSearch/pull/17890)
- [17907](https://github.com/opensearch-project/OpenSearch/pull/17907)
- [17941](https://github.com/opensearch-project/OpenSearch/pull/17941)
- [18043](https://github.com/opensearch-project/OpenSearch/pull/18043)
- [18054](https://github.com/opensearch-project/OpenSearch/pull/18054)
- [18069](https://github.com/opensearch-project/OpenSearch/pull/18069)
- [18085](https://github.com/opensearch-project/OpenSearch/pull/18085)
- [18102](https://github.com/opensearch-project/OpenSearch/pull/18102)
- [18107](https://github.com/opensearch-project/OpenSearch/pull/18107)
- [18155](https://github.com/opensearch-project/OpenSearch/pull/18155)
- [18161](https://github.com/opensearch-project/OpenSearch/pull/18161)
- [18201](https://github.com/opensearch-project/OpenSearch/pull/18201)
- [18212](https://github.com/opensearch-project/OpenSearch/pull/18212)
- [18243](https://github.com/opensearch-project/OpenSearch/pull/18243)
- [18250](https://github.com/opensearch-project/OpenSearch/pull/18250)
- [18327](https://github.com/opensearch-project/OpenSearch/pull/18327)
- [18346](https://github.com/opensearch-project/OpenSearch/pull/18346)
- [18351](https://github.com/opensearch-project/OpenSearch/pull/18351)

For more details on the failed tests refer to [OpenSearch Gradle Check Metrics](https://metrics.opensearch.org/_dashboards/app/dashboards#/view/e5e64d40-ed31-11ee-be99-69d1dbc75083) dashboard.

Might be a bug with a floating point comparison:

```
REPRODUCE WITH: ./gradlew ':client:rest-high-level:asyncIntegTest' --tests "org.opensearch.client.SearchIT.testSearchWithMatrixStats" -Dtests.seed=630A15BDCB6BB5D9 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=ksb -Dtests.timezone=Europe/Jersey -Druntime.java=21

SearchIT > testSearchWithMatrixStats FAILED
    java.lang.AssertionError: expected:<1830.0000000000002> but was:<1829.9999999999998>
        at __randomizedtesting.SeedInfo.seed([630A15BDCB6BB5D9:1301905AE5084B97]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:555)
        at org.junit.Assert.assertEquals(Assert.java:685)
        at org.opensearch.client.SearchIT.testSearchWithMatrixStats(SearchIT.java:570)
```

I'm seeing this flaky a lot on my unrelated PR(s).

```
java.lang.AssertionError: expected:<1830.0000000000002> but was:<1830.0>
```

```
java.lang.AssertionError: expected:<1830.0000000000002> but was:<1830.0000000000005>
```

There are several tests which provide a `0d` tolerance interval.

--- a/client/rest-high-level/src/test/java/org/opensearch/client/SearchIT.java
+++ b/client/rest-high-level/src/test/java/org/opensearch/client/SearchIT.java
@@ -567,12 +567,12 @@ public class SearchIT extends OpenSearchRestHighLevelClientTestCase {
         MatrixStats matrixStats = searchResponse.getAggregations().get("agg1");
         assertEquals(5, matrixStats.getFieldCount("num"));
         assertEquals(56d, matrixStats.getMean("num"), 0d);
-        assertEquals(1830.0000000000002, matrixStats.getVariance("num"), 0d);
+        assertEquals(1830.0000000000002, matrixStats.getVariance("num"), 1.0e-12);
         assertEquals(0.09340198804973039, matrixStats.getSkewness("num"), 0d);
         assertEquals(1.2741646510794589, matrixStats.getKurtosis("num"), 0d);
         assertEquals(5, matrixStats.getFieldCount("num2"));
         assertEquals(29d, matrixStats.getMean("num2"), 0d);
-        assertEquals(330d, matrixStats.getVariance("num2"), 0d);
+        assertEquals(330d, matrixStats.getVariance("num2"), 1.0e-12);
         assertEquals(-0.13568039346585542, matrixStats.getSkewness("num2"), 1.0e-16);
         assertEquals(1.3517561983471071, matrixStats.getKurtosis("num2"), 0d);
         assertEquals(-767.5, matrixStats.getCovariance("num", "num2"), 0d);

